src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -206,8 +218,19 @@ private String getArtifactDownloadURL(ComponentIdentifier componentIdentifier, S
             }, "get-artifact-size", logger);
         } catch (InterruptedException e) {
             throw e;
+        } catch (GreengrassV2DataException e) {
+            if (e.statusCode() == HttpStatusCode.FORBIDDEN) {
+                throw new PackageDownloadException(getErrorString("Failed to get the pre-signed url of a public or a "
+                        + "Lambda component artifact. GetComponentVersionArtifact returns 403 Access Denied"),
```

Comment:
list the permission which is needed to execute this correctly, and note that it is in the iot policy.

Reply:
will update